src/peerconnection.cc

```diff
@@ -34,27 +34,112 @@ Persistent<Function> PeerConnection::constructor;
   _setRemoteDescriptionObserver = new rtc::RefCountedObject<SetRemoteDescriptionObserver>( this );
 
-  // FIXME: don't hardcode this, read from args instead
-  webrtc::PeerConnectionInterface::IceServer iceServer;
-  iceServer.uri = "stun:stun.l.google.com:19302";
-  _iceServers.push_back(iceServer);
+  if (!configuration.IsEmpty()) {
+    v8::Local<v8::Value> iceservers_value = configuration->Get(NanNew("iceServers"));
+    
+    if (!iceservers_value.IsEmpty() && iceservers_value->IsArray()) {
+      v8::Local<v8::Array> list = v8::Local<v8::Array>::Cast(iceservers_value);
 
-  webrtc::FakeConstraints constraints;
-  constraints.AddOptional(webrtc::MediaConstraintsInterface::kEnableDtlsSrtp, webrtc::MediaConstraintsInterface::kValueTrue);
+      for (unsigned int index = 0; index < list->Length(); index++) {
+        v8::Local<v8::Value> server_value = list->Get(index);
+
+        if (!server_value.IsEmpty() && server_value->IsObject()) {
+          v8::Local<v8::Object> server = v8::Local<v8::Object>::Cast(server_value);
+
+          v8::Local<v8::Value> url_value = server->Get(NanNew("url"));
+          v8::Local<v8::Value> username_value = server->Get(NanNew("username"));
+          v8::Local<v8::Value> credential_value = server->Get(NanNew("credential"));
+
+          if (!url_value.IsEmpty() && url_value->IsString()) {
+            webrtc::PeerConnectionInterface::IceServer entry;
+            v8::String::Utf8Value url(url_value->ToString());
+
+            entry.uri = *url;
+
+            if (!username_value.IsEmpty() && username_value->IsString()) {
+              v8::String::Utf8Value username(username_value->ToString());
+              entry.username = *username;
+            }
+
+            if (!credential_value.IsEmpty() && credential_value->IsString()) {
+              v8::String::Utf8Value credential(credential_value->ToString());
+              entry.password = *credential;
+            }
+
+            _iceServers.push_back(entry);
+          }
+        }        
+      }
+    }
+  }
+
+  bool enableSecure = true;
+  bool enableAudio = false;
+  bool enableVideo = false;
+  
+  if (!constraints.IsEmpty()) {
+    v8::Local<v8::Value> optional_value = constraints->Get(NanNew("optional"));
+    
+    if (!optional_value.IsEmpty() && optional_value->IsArray()) {
+      v8::Local<v8::Array> options = v8::Local<v8::Array>::Cast(optional_value);
+      
+      for (unsigned int index = 0; index < options->Length(); index++) {
+        v8::Local<v8::Value> option_value = options->Get(index);
+        
+        if (!option_value.IsEmpty() && option_value->IsObject()) {
+          v8::Local<v8::Object> option = v8::Local<v8::Object>::Cast(option_value);
+          v8::Local<v8::Value> DtlsSrtpKeyAgreement = option->Get(NanNew("DtlsSrtpKeyAgreement"));
+          
+          if (!DtlsSrtpKeyAgreement.IsEmpty() && DtlsSrtpKeyAgreement->IsFalse()) {
+            enableSecure = false;
+          }
+        }
+      }
+    }
+/*
```

Comment:
This should be removed instead of commented out. If we need it again, it's in the repo history.